pkg/controller/storagecluster/update.go

```diff
@@ -94,9 +94,10 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 		}
 
 		// check if pod is running in a node which has internal kvdb running in it
-		if _, isKvdbNode := kvdbNodes[pod.Spec.NodeName]; cluster.Spec.Kvdb != nil && cluster.Spec.Kvdb.Internal && isKvdbNode {
-			// if number of unavailable kvdb nodes is greater than or equal to 1, then dont restart portworx on this node
-			if numUnavailableKvdb > 0 {
+		if _, isKvdbNode := kvdbNodes[pod.Spec.NodeName]; err == nil && cluster.Spec.Kvdb != nil && cluster.Spec.Kvdb.Internal && isKvdbNode {
+			// if number of unavailable kvdb nodes is greater than or equal to 1, or lesser than 3 entries are present in the kvdb map
+			// then dont restart portworx on this node
+			if numUnavailableKvdb > 0 || len(kvdbNodes) < 3 {
 				logrus.Infof("Number of unavaliable KVDB members exceeds 1, temporarily skipping update for this node to prevent KVDB from going out of quorum ")
```

Comment:
update the comment here based on which usecase it is, numUnavailableKvdb > 0 or len(kvdbNodes) < 3